src/backend/common/helpers/district_helper.py

```diff
@@ -351,8 +351,120 @@ def _get_rookie_bonus(cls, year: Year, team_rookie_year: Year) -> int:
 
         return bonus
 
+    @classmethod
+    def _get_alliance_number_from_teams(
+        cls,
+        alliance_selections: List[EventAlliance],
+        teams: List[TeamKey],
+    ):
+        search_team = teams[0]
+        for num, alliance in enumerate(alliance_selections):
+            if search_team in alliance["picks"]:
+                return num
+
+        return None
+
     @classmethod
     def _calc_elim_match_points(
+        cls,
+        district_points: EventDistrictPoints,
+        matches: List[Match],
+        alliance_selections: List[EventAlliance],
+        POINTS_MULTIPLIER: int,
+    ):
+        # match_set_key -> alliance -> num wins
+        elim_num_wins: DefaultDict[str, DefaultDict[AllianceColor, int]] = defaultdict(
+            lambda: defaultdict(int)
+        )
+
+        elim_alliance_matches_played: DefaultDict[int, int] = defaultdict(int)
+        elim_team_matches_played: DefaultDict[TeamKey, int] = defaultdict(int)
+
+        for match in matches:
+            if not match.has_been_played or match.winning_alliance == "":
+                # Skip unplayed matches
+                continue
+
+            match_set_key = "{}_{}{}".format(
+                match.event_key_name, match.comp_level, match.set_number
+            )
+            winning_alliance = cast(AllianceColor, match.winning_alliance)
+            losing_allaince = cast(AllianceColor, match.losing_alliance)
+            elim_num_wins[match_set_key][winning_alliance] += 1
+
+            # Get alliance numbers
+            winning_alliance_number = cls._get_alliance_number_from_teams(
+                alliance_selections,
+                match.alliances[winning_alliance]["teams"],
+            )
+            losing_alliance_number = cls._get_alliance_number_from_teams(
+                alliance_selections,
+                match.alliances[losing_allaince]["teams"],
+            )
+
+            elim_alliance_matches_played[winning_alliance_number] += 1
+            elim_alliance_matches_played[losing_alliance_number] += 1
+
+            teams = (
+                match.alliances[winning_alliance]["teams"]
+                + match.alliances[losing_allaince]["teams"]
+            )
+
+            for team in teams:
+                elim_team_matches_played[team] += 1
+
+            # Loser of match 12 receives 7 points
+            if match.comp_level == CompLevel.SF and match.set_number == 12:
+                for team in alliance_selections[losing_alliance_number]["picks"]:
+                    multiplier = (
+                        POINTS_MULTIPLIER
+                        * elim_team_matches_played[team]
+                        / elim_alliance_matches_played[losing_alliance_number]
```

Comment:
Hey @andrewda @fangeugene - there's a small bug here. This should be based on the fraction of elim matches WON, not the fraction of elim matches played. Here's the rule:
```
If a team does not play 100% of the Playoff MATCHES won by their ALLIANCE, their Playoff Performance
points equals their ALLIANCE Advancement points multiplied by the percentage of Playoff MATCHES
**won** by their ALLIANCE in which that team was a participant
```

This is affecting frc1672 in the Mt Olive distict event. 

They should have `ceil(2/3*13)=9` and instead TBA is showing them with `ceil(3/5*13)=8`.

Reply:
Thanks for the heads up, working on a fix now.